lib/rules/no-unused-vars.js

```diff
@@ -102,19 +122,12 @@ module.exports = function(context) {
             }
         },
 
+        "JSXExpressionContainer": function(node) {
+            flagJSXAsUsed(node.expression.name);
```

Comment:
If I am correct in this cases the code between the `{}` is parsed as plain JavaScript and we fall back to the general case.

I did not get any problem with the above examples, the defined and undefined variables are flagged correctly.

Did you find any issue ?